File: Java/Spring/e-commerce/src/main/java/com/ecommerce/ecommerce/entities/concretes/Product.java
package com.ecommerce.ecommerce.entities.concretes;
import com.ecommerce.ecommerce.core.entities.IEntity;
import lombok.Data;
import javax.persistence.*;

@Table(name="products")
@Entity
@Data
public class Product implements IEntity {
    @Id
    @GeneratedValue
    @Column(name="id")
    private int id;
    @Column(name="category_id")
    private int categoryId;
    @Column(name="supplier_id")
    private int supplierId;
    @Column(name="product_name")
    private String productName;
    @Column(name="product_desc")
    private String productDescription;
    @Column(name="unit_price")
    private double unitPrice;
    @Column(name="units_in_stock")
    private double unitsInStock;


}

File: Kotlin/ecommerce/src/main/kotlin/com/semih/ecommerce/entities/concrete/Product.kt
package com.semih.ecommerce.entities.concrete

import javax.persistence.*

@Entity
@Table(name="products")
data class Product(
    @Id @GeneratedValue @Column(name="id") val id:Long?,
    @Column(name = "category_id") val categoryId: Long,
    @Column(name="supplier_id") val supplierId:Long,
    @Column(name="product_name") val productName:String,
    @Column(name="product_desc") val productDescription:String,
    @Column(name="unit_price") val unitPrice:Double,
    @Column(name="units_in_stock") val unitsInStock:Double
)
